File: backend/src/main/java/com/accesodatos/entity/CartItem.java
package com.accesodatos.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(exclude = "user")
@ToString(exclude = "user")
@Table(name = "cart_items")
public class CartItem {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long cartId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    @JsonBackReference("user-cart")
	private UserEntity user;
	
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "product_id")
    @JsonBackReference("product-cart")
	private Product product;
	
	private int cuantity;
}

File: frontend/src/types/cartItem.ts
import { ProductCartItemResponseDto, ProductResponse } from "./Product";
import { UserResponseDto } from "../types/User";

export interface CartItemRequestDto {
  userId: number;
  productId: number; // Long se mapea a number en TS
  cuantity: number;
}

export interface CartItemResponseDto {
  cartId: number; // Long se mapea a number
  user: UserResponseDto;
  product: ProductResponse;
  cuantity: number;
}

export interface CartItemUserResponseDto {
  product: ProductCartItemResponseDto;
  cuantity: number;
}
